src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -370,6 +371,40 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
                     }
                 }
             }
+            else if (AdvSimd.IsSupported)
```

Comment:
This method is getting pretty big. Is the JIT going to have trouble optimizing it due to size/number of locals, even with the dead code elimination?

Reply:
Good point. I don't see any difference like that though. See attached generated JIT code in PR description.